source/extensions/filters/common/lua/wrappers.cc

```diff
@@ -125,6 +127,85 @@ int MetadataMapWrapper::luaPairs(lua_State* state) {
   return 1;
 }
 
+int RequestInfoWrapper::luaDynamicMetadata(lua_State* state) {
+  if (metadata_wrapper_.get() != nullptr) {
+    metadata_wrapper_.pushStack();
+  } else {
+    metadata_wrapper_.reset(DynamicMetadataMapWrapper::create(state, request_info_), true);
+  }
+  return 1;
+}
+
+int RequestInfoWrapper::luaProtocol(lua_State* state) {
+  switch (request_info_.protocol().value()) {
```

Comment:
I have not looked at the entire code. But one quick comment here, this whole `switch` can be replaced with this [method call](https://github.com/envoyproxy/envoy/blob/master/source/common/http/utility.cc#L328)

Reply:
Gotcha! Thanks @ramaraochavali!